app/model/RuleContext.scala

```diff
@@ -17,19 +17,25 @@
 package model
 
 import connector._
+import play.api.{Logger, LoggerLike}
 import uk.gov.hmrc.play.frontend.auth.AuthContext
 import uk.gov.hmrc.play.http.HeaderCarrier
 import uk.gov.hmrc.play.http.logging.MdcLoggingExecutionContext.fromLoggingDetails
 
 import scala.concurrent.Future
 
 case class RuleContext(authContext: AuthContext)(implicit hc: HeaderCarrier) {
+  val logger: LoggerLike = Logger
   val selfAssessmentConnector: SelfAssessmentConnector = SelfAssessmentConnector
   val frontendAuthConnector: FrontendAuthConnector = FrontendAuthConnector
   val userDetailsConnector: UserDetailsConnector = UserDetailsConnector
 
   lazy val userDetails = currentCoAFEAuthority.flatMap { authority =>
-    userDetailsConnector.getUserDetails(authority.userDetailsLink)
+    authority.userDetailsLink.map(userDetailsConnector.getUserDetails).getOrElse {
+      val exception = new RuntimeException("userDetailsLink is not defined")
+      logger.warn("failed to get user details", exception)
```

Comment:
what is the value of logging an exception that we manually create therefore we don't have any stacktrace or so?